src/plugins/dkommon/dkommon.cpp

```diff
@@ -299,3 +431,18 @@ dkommon::~dkommon()
 {
     delete[] offsets;
 }
+
+bool dkommon::stop()
+{
+    if (!is_stopping() && !done_final_analysis)
+    {
+        m_is_stopping = true;
+        PRINT_DEBUG("[dkommon] Injecting KiDeliverApc\n");
+        // Hook dummy function so we could make final system analysis
+        breakpoint_in_system_process_searcher bp;
+        register_trap(nullptr, final_check_cb, bp.for_syscall_name("KiDeliverApc"));
```

Comment:
Why do you need a new trap added here instead of just performing the final check?

Reply:
The plugin receives the drakvuf context in the constructor. You can just store it in the dkommon class for later use by the stop function.

Reply:
In fact, since you are extending the `pluginex` class now you already have it stored for you in `this->drakvuf`.

Reply:
But if I do my analysis in `stop` function I wont have access to `drakvuf_trap_info_t` context which is required for `fmt::print` function?
Imo it is ok to hook `KiDeliverApc` since its called almost instantly.